Add size message checks for logging

pvollan opened the pull request:
#### 3a0860459e01ee48e7a176581353ca254d3f9915
<pre>
Add size message checks for logging
<a href="https://bugs.webkit.org/show_bug.cgi?id=293558">https://bugs.webkit.org/show_bug.cgi?id=293558</a>
<a href="https://rdar.apple.com/152010794">rdar://152010794</a>

Reviewed by Sihui Liu.

* Source/WebKit/Shared/LogStream.cpp:
(WebKit::LogStream::logOnBehalfOfWebContent):
* Source/WebKit/Shared/LogStream.h:
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::registerLogHook):

Canonical link: <a href="https://commits.webkit.org/296808@main">https://commits.webkit.org/296808@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/63eb21aac340e9e981ff05689460f8e6d491cc26

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/109671 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/29329 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/19758 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/115692 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/59905 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/30007 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/37917 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/83338 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/112619 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/23910 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/98766 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/63797 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/23291 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/59486 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/93285 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/16948 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/118484 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/36710 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/27184 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/92346 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/37083 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/95025 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/92167 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/37125 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/14872 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/32538 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17695 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/36604 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/42075 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/36265 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/39607 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/37974 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -856,12 +856,18 @@ static bool shouldIgnoreLogMessage(const char* logSubsystem)
             auto logString = unsafeSpan8IncludingNullTerminator(messageString);
-            WebProcess::singleton().sendLogOnStream(logChannel, logCategory, logString, type);
+            if (logString.size() < logStringMaxSize)
+                WebProcess::singleton().sendLogOnStream(logChannel, logCategory, logString, type);
```
Could this affect WebRTC logging which can be quite long (but quite useful for debugging)?
Also, if a channel log is debug, I would expect logging to be high in general.

pvollan replied:
That is a good point! Is there WebRTC logging in the WebContent process? If so, would you know the approximate size of the largest log string?

Thanks for reviewing!

pvollan replied:
In the latest patch, I increased the max size of the log string to accommodate for this.

szewai commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -856,12 +856,18 @@ static bool shouldIgnoreLogMessage(const char* logSubsystem)
         if (char* messageString = os_log_copy_message_string(msg)) {
             auto logString = unsafeSpan8IncludingNullTerminator(messageString);
-            WebProcess::singleton().sendLogOnStream(logChannel, logCategory, logString, type);
+            if (logString.size() < logStringMaxSize)
```
Should we truncate the string and send instead of dropping message?

pvollan replied:
That is a good point, will fix!

Thanks for reviewing!

szewai commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -856,12 +856,22 @@ static bool shouldIgnoreLogMessage(const char* logSubsystem)
+                String logStringTruncated = String::fromUTF8(logString).left(logStringMaxSize - 1);
+                logString = byteCast<uint8_t>(logStringTruncated.utf8().spanIncludingNullTerminator());
+                WebProcess::singleton().sendLogOnStream(logChannel, logCategory, logString, type);
+            } else
```
Maybe we could move `WebProcess::singleton().sendLogOnStream(` out of `if` condition and remove `else`

pvollan replied:
That is a good point, will fix.

Thanks for reviewing!

pvollan committed:
Add size message checks for logging
https://bugs.webkit.org/show_bug.cgi?id=293558
rdar://152010794

Reviewed by Sihui Liu.

* Source/WebKit/Shared/LogStream.cpp:
(WebKit::LogStream::logOnBehalfOfWebContent):
* Source/WebKit/Shared/LogStream.h:
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::registerLogHook):

Canonical link: https://commits.webkit.org/296808@main
